### PR TITLE
Refactor the project folder structure

### LINKED ISSUE #236: File structure refactor proposal
I've been dreaming about some slight changes to the folder structure. A few motivations:

* clean up the root folder so we don't have legit source files astrewn with our dotfiles.
* codify some of the conventions we've found
* make the structure a bit more newb-friendly

Here's what i'm proposing

```md
- test/
- closure/
- typings/
- scripts/
  - launch_chrome.js
  - run-smoke-tests.sh
  - download_chrome.sh
- src/
  - index.js
  - scheduler.js
  - aggregator.js
  - auditor.js
  - runner.js
  - cli/
    - index.js
    - printer.js
  - lib/  (née helpers)
    - drivers/
  - gatherers/
  - audits/
  - aggregators/
  - metrics/
- extension/
- report/
- index.js (empty-ish)
- (all the dotfiles)
```

* CLI would now be via `node src/cli https://pwa.rocks.com` or continue to use `npm link` & `lighthouse`
* index.js would be the entry point that's shared across both CLI & extension. (it's close to what it is today, mostly deferring to runner.js)

Thoughts?

### PR DESCRIPTION
**Pull request by paulirish**
Applies the refactor sketched out in #236

##### Significant changes:

* most source files now held within `/src/`. You can think of `src` as lighthouse-core. Any printers or reporters specific to extension/cli belong outside of it.
* `/cli/` is top-level, as it's siblings with `extension/`
* `helpers/` is now `lib/`
* `gather-scheduler` is now just `scheduler`
* `runner.js` is now `lighthouse.js`
* `/scripts/` holds all various bash & python scripts
* Running the CLI is different, but now three keystrokes easier. From `node cli.js` --> `node cli`

Fixes #236

### CODE REVIEW

File: src/gatherers/accessibility.js [hunk 1]
```diff
@@ -21,7 +21,7 @@
 const Gather = require('./gather');
 const fs = require('fs');
 const path = require('path');
-const axe = fs.readFileSync(path.join(__dirname, '../node_modules/axe-core/axe.min.js'));
+const axe = fs.readFileSync(path.join(__dirname, '../../node_modules/axe-core/axe.min.js'));
```
Comment (brendankenny) [inline src/gatherers/accessibility.js]: wait, did we not end up `require.resolve()`ing this?

**Comment by brendankenny** [comment]
LGTM. One thing that's not brought up in current #236: tests should probably moved to reflect the directory structure of the files they're testing.

File: src/gatherers/accessibility.js (hunk 1 above)
Comment (brendankenny) [inline src/gatherers/accessibility.js]: ah, we missed a few things in the great PR purge of April 23 :P

Comment (paulirish) [inline src/gatherers/accessibility.js]: Yup. I'm happy to add that as a followup today. I just wanted to do bare-minimum changes in this particular PR.

**Comment by paulirish** [comment]
> tests should probably moved to reflect the directory structure of the files they're testing. 

Good call! I just started work on that. There's enough changes that we should keep it as a separate PR. I'll have it up soon.

Any other feelings on this PR? Want to get it in place soon to not hold anyone else up.

**Comment by brendankenny** [comment]
> Any other feelings on this PR? Want to get it in place soon to not hold anyone else up.

no, LGTM if you want to move that to a separate PR

**Comment by samccone** [comment]
squash and merge +1

**Pull request merged**